basics-of-csharp/basics-of-csharp.md

```diff
@@ -0,0 +1,578 @@
+
+  ```csharp
+  using System.Console;
+
+  ...
+
+  // Console belongs to the System namespace
+  Console.WriteLine();
+  ```
+
+2. To allow you to access static members of a type without having to qualify the access with the type name:
+
+  ```csharp
+  using static System.Math;
+
+  ...
+
+  // Abs belongs to the System.Math namespace
+  Abs(-3.14);
+  ```
+
+3. To create an alias for a namespace or a type. This is called a *using alias directive*.
+
+  ```csharp
+  using Project = PC.MyCompany.Project;
+  ```
+
+
+#### namespace
+
+The namespace keyword is used to declare a `scope` that contains a set of related objects. You can use a namespace to organize code elements and to create globally unique types.
+
+You can declare the following types within a namespace:
+- another namespace
+- class
+- interface
+- struct
+- enum
+- delegate
+
+If you do not declare a namespace the compiler will assign one for you.
+
+In the context of our SampleConsoleApp, the class `Program` is available under the `ConsoleApplication` namespace.
+
+```csharp
+namespace ConsoleApplication
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            Console.WriteLine("Hello World!");
+        }
+    }
+}
+```
+
+You could create instances of `ConsoleApplication.Program` by importing it with `using ConsoleApplication.Program`
+
+### Main Class
+
+The next part of our program is the class with the `Main` method
+
+```csharp
+...
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        Console.WriteLine("Hello World!");
+    }
+}
+
+```
+
+Classes let you create your own objects by grouping together variables and methods.
+
+Every C# application must contain a single `Main` method specifying where program execution is to begin. In C#, `Main` is capitalized, while Java uses lowercase main.
+
+
+```csharp
+public static void Main(string[] args)
+{
+    ...
+}
+```
+
+
+Main can only return int or void, and has an optional string array argument to represent command-line parameters
+
+I could pass my friend's name "Andy" as a command-line parameter like this:
+
+```sh
+dotnet run Andy
+```
+
+```csharp
+public static void Main(string[] args)
+{
+    string name;
+    try {
+      name = args[0]; // Andy
+    } catch (Exception e) {
+      // if the user fails to pass a name just use 'World'
+      name = "World";
+    }
+    Console.WriteLine("Hello " + name); // Hello Andy
+}
+```
+
+
+### Comments
+
+You may want to leave a comment for yourself or other developers to remember important things.
+
+```chsarp
+// this is a helpful single-line comment
+
+/*
+  This is a
+  helpful comment
+  on multiple
+  lines ...
+ */
+```
+
+### Data Types and Variables
+
+Data types help you define what kind of value a variable can contain.
+
+
+#### Primitive Types
+
+- bool (true/false)
+- byte (unsigned 8-bit integer range 0 to 255)
+- char ('a')
+- short, int, long
+- float, double, decimal
```

Comment:
- `bool` (true/false)
- `byte` (unsigned 8-bit integer range 0 to 255)
- `char` ('a')
- `short`, `int`, `long`
- `float`, `double`, `decimal`